library/nxos_feature.py

```diff
@@ -0,0 +1,298 @@
+        description:
+            - Dictates connection protocol to use for NX-API
+        required: false
+        default: http
+        choices: ['http', 'https']
+        aliases: []
+'''
+
+EXAMPLES = '''
+# Ensure lacp is enabled
+- nxos_feature: feature=lacp state=enabled host={{ inventory_hostname }}
+
+# Ensure ospf is disabled
+- nxos_feature: feature=ospf state=disabled host={{ inventory_hostname }}
+
+# Ensure vpc is enabled
+- nxos_feature: feature=vpc state=enabled host={{ inventory_hostname }}
+
+'''
+
+RETURN = '''
+
+proposed:
+    description: proposed feature state
+    returned: always
+    type: dict
+    sample: {"state": "disabled"}
+existing:
+    description: existing state of feature
+    returned: always
+    type: dict
+    sample: {"state": "enabled"}
+end_state:
+    description: feature state after executing module
+    returned: always
+    type: dict
+    sample: {"state": "disabled"}
+state:
+    description: state as sent in from the playbook
+    returned: always
+    type: string
+    sample: "disabled"
+commands:
+    description: command string sent to the device
+    returned: always
+    type: string
+    sample: "no feature eigrp ;"
+changed:
+    description: check to see if a change was made on the device
+    returned: always
+    type: boolean
+    sample: true
+'''
+
+import socket
+import xmltodict
+try:
+    HAS_PYCSCO = True
+    from pycsco.nxos.device import Device
+    from pycsco.nxos.device import Auth
+    from pycsco.nxos.error import CLIError
+except ImportError as ie:
+    HAS_PYCSCO = False
+
+
+def parsed_data_from_device(device, command, module):
+    try:
+        data = device.show(command)
+    except CLIError as clie:
+        module.fail_json(msg='Error sending {}'.format(command),
+                         error=str(clie))
+
+    data_dict = xmltodict.parse(data[1])
+    body = data_dict['ins_api']['outputs']['output']['body']
+
+    return body
+
+
+def apply_key_map(key_map, table):
+    new_dict = {}
+    for key, value in table.items():
+        new_key = key_map.get(key)
+        if new_key:
+            new_dict[new_key] = str(value)
+    return new_dict
+
+
+def apply_value_map(value_map, resource):
+    for key, value in value_map.items():
+        resource[key] = value[resource.get(key)]
+    return resource
+
+
+def command_list_to_string(command_list):
+    """Converts list of commands into proper string for NX-API
+
+    Args:
+        cmds (list): ordered list of commands
+
+    Returns:
+        str: string of commands separated by " ; "
+
+    """
+    if command_list:
+        command = ' ; '.join(command_list)
+        return command + ' ;'
+    else:
+        return ''
+
+
+def get_feature_state(available_features, feature):
+    if 'enabled' in available_features[feature]:
+        existstate = 'enabled'
+    else:
+        existstate = 'disabled'
+    return existstate
+
+
+def get_features(device, feature, module):
+    available_features = {}
```

Comment:
Actually I coded it in a way to have something like:

    available_features = {
        "ospf" : "enabled",
        "pim" : "disabled",
        ...
        }

Is it not good for you?